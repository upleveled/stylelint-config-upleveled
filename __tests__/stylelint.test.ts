import { $ } from 'execa';
import { expect, test } from 'vitest';

test('run stylelint and expect errors', async () => {
  try {
    const stylelintOutput =
      await $`pnpm stylelint '**/*.{css,scss,less,js,tsx}'`;
    if (stylelintOutput.exitCode === 0) {
      throw new Error('No stylelint errors thrown');
    }
  } catch (error) {
    expect(error).toMatchSnapshot();
  }
});
