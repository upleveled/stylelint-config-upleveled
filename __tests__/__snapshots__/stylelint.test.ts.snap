// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`run stylelint and expect errors 1`] = `
[Error: Command failed with exit code 2: pnpm stylelint '**/*.{css,scss,less,js,tsx}'

__tests__/index.module.scss
  1:1   ✖  Unexpected unknown type selector "lol"         selector-type-no-unknown
  1:5   ✖  Unexpected empty block                         block-no-empty
  5:3   ✖  Unexpected unknown property "backgound-color"  property-no-unknown
  7:3   ✖  Unexpected duplicate "--custom-property"       declaration-block-no-duplicate-custom-properties
 16:9   ✖  Unexpected unknown type selector "c"           selector-type-no-unknown
 16:13  ✖  Unexpected unknown type selector "d"           selector-type-no-unknown
 16:15  ✖  Unexpected unknown type selector "e"           selector-type-no-unknown

7 problems (7 errors, 0 warnings)
]
`;
